### PR TITLE
Improve Unity build script

### PR DESCRIPTION
**Pull request by annakocheshkova**
* [X] Are the files formatted correctly?
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change? 

[AB#71834](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/71834)

### CODE REVIEW

**Comment by annakocheshkova** [comment]
Tested all the cases.

File: build.cake [hunk 1]
```diff
@@ -298,6 +306,11 @@ Task("BuildAndroidContentProvider").Does(()=>
     {
         appName = "BreakpadSupport";
         BuildAndroidLibrary(appName, "", false);
+        Warning ("Ignoring BreakpadSupport build failure... It's ok.");
```
Comment (guperrot) [inline build.cake]: Can you explain that again, it works but it fails, I never understood (and how can we fix it for real).

Comment (annakocheshkova) [reply build.cake]: No idea, I've searched many forums while implementing that, it just gives an error while the file is being present. My assumption was that it had something to do with relative path 2 levels up from the current folder. Anyway later in the code I make a real check if the `.so` is there so it should be ok.

Comment (guperrot) [reply build.cake]: Ok we might want to improve the warning and/or add a comment before to explain.

**Review by MatkovIvan — DISMISSED** [review]

File: test-tools.cake [hunk 2]
```diff
@@ -161,7 +161,11 @@ Task("CreateIosArchive").IsDependentOn("IncreaseIosVersion")
     var xcodeProjectPath = GetDirectories(BuildFolder + "/*/*.xcodeproj").Single().FullPath;
     Information("Creating archive...");
     var archiveName = Statics.TemporaryPrefix + "iosArchive.xcarchive";
-    StartProcess("xcodebuild", "-project \"" + xcodeProjectPath + "\" -configuration Release -scheme Unity-iPhone -archivePath \"" + archiveName + "\" archive");
+    var result = StartProcess("xcodebuild", "-project \"" + xcodeProjectPath + "\" -configuration Release -scheme Unity-iPhone -archivePath \"" + archiveName + "\" archive");
+    if (result != 0)
+    {
+        throw new Exception("Failed to archive ipa.");
```
Comment (achocron) [inline test-tools.cake]: Can you include the result code in the message?

File: test-tools.cake [hunk 3]
```diff
@@ -171,9 +175,13 @@ Task("CreateIosArchive").IsDependentOn("IncreaseIosVersion")
         "<plist version=\"1.0\"><dict></dict></plist>");
     Information("Creating ipa...");
     DeleteFileIfExists(CurrentApp.AppPath);
-    StartProcess("xcodebuild", "-exportArchive -archivePath \"" + archiveName +
+    result = StartProcess("xcodebuild", "-exportArchive -archivePath \"" + archiveName +
                     "\" -exportPath \"" + CurrentApp.AppPath +
                     "\" -exportOptionsPlist \"" + plistName + "\"");
+    if (result != 0)
+    {
+        throw new Exception("Failed to export ipa.");
```
Comment (achocron) [inline test-tools.cake]: Result code in message?

File: build.cake [hunk 4]
```diff
@@ -410,7 +440,11 @@ Task("Externals-Unity-Packages").Does(()=>
         DownloadFile(package.Url, destination);
         Information("Importing package " + package.Name + ". This could take a minute.");
         var command = package.AdditionalImportArgs + " -importPackage " + destination.FullPath;
-        ExecuteUnityCommand(command);
+        var result = ExecuteUnityCommand(command);
+        if (result != 0)
+        {
+            throw new Exception("Something went wrong while importing packages.");
```
Comment (achocron) [inline build.cake]: Can you add result code in message?

File: build.cake [hunk 5]
```diff
@@ -396,7 +422,11 @@ Task ("Externals-Uwp-IL2CPP-Dependencies")
 
         // Process UWP IL2CPP dependencies.
         Information ("Processing UWP IL2CPP dependencies. This could take a minute.");
-        ExecuteUnityCommand ("-executeMethod AppCenterPostBuild.ProcessUwpIl2CppDependencies");
+        var result = ExecuteUnityCommand ("-executeMethod AppCenterPostBuild.ProcessUwpIl2CppDependencies");
+        if (result != 0)
+        {
+            throw new Exception("Something went wrong while executing post build script");
```
Comment (achocron) [inline build.cake]: Can you add result code in message?

File: build.cake [hunk 6]
```diff
@@ -312,9 +328,19 @@ void BuildAndroidLibrary(string appName, string libraryName, bool copyBinary = t
...
 
     if (copyBinary) {
+
+        // We check the result of the build only here because we need to check build
+        // result only in case we build native binary.
+        // In another case, when we build breakpad, we don't check it
+        // Due to the known issue that breakpad build always reports false failure.
+        if (result != 0)
+        {
+            throw new Exception("Failed to build android native library.");
```
Comment (achocron) [inline build.cake]: Can you add result code in message?

**Review by achocron — CHANGES_REQUESTED** [review]

**Review by achocron — APPROVED** [review]

**Pull request merged**